core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/dataset/DatasetResource.scala

```diff
@@ -175,7 +175,7 @@ object DatasetResource {
   def resolveFilePath(
       filePath: java.nio.file.Path
   ): (Dataset, DatasetVersion, java.nio.file.Path) = {
-    val pathSegments = filePath.toString.stripPrefix("/").split("/")
```

Comment:
Can you try to use the native method of `java.nio.file.Path` like the below:
(0 until filePath.getNameCount).map(filePath.getName).map(_.toString)

Reply:
Of course. Do I need to modify the following if statement to if (pathSegments.length != 4)?

Reply:
> Can you try to use the native method of `java.nio.file.Path` like the below: (0 until filePath.getNameCount).map(filePath.getName).map(_.toString)

Of course. Do I need to modify the following if statement to if (pathSegments.length != 4)?